src/main/java/net/sf/jabref/logic/net/URLDownload.java

```diff
@@ -119,6 +118,20 @@ private URLConnection openConnection() throws IOException {
             }
 
         }
+
+        // normally, 3xx is redirect
+        int status = connection.getResponseCode();
+        if (status != HttpURLConnection.HTTP_OK) {
+            if (status == HttpURLConnection.HTTP_MOVED_TEMP
+                    || status == HttpURLConnection.HTTP_MOVED_PERM
+                    || status == HttpURLConnection.HTTP_SEE_OTHER) {
+                // get redirect url from "location" header field
+                String newUrl = connection.getHeaderField("Location");
+                // open the new connnection again
+                connection = (HttpURLConnection) new URLDownload(newUrl).openConnection();
```

Comment:
Can this be tested with mockito?

Can't it happen that the new location also points to a new location?

The client library doesn't transparently redirects and there is no utility layer in the lib? Strange....